tap-agent/src/agent/sender_account.rs

```diff
@@ -484,12 +493,23 @@ impl Actor for SenderAccount {
                     }
                 }
 
-                // Maybe allow the sender right after the potential RAV request. This way, the
-                // sender can be allowed again as soon as possible if the RAV was successful.
-
-                let should_allow = state.denied && !state.deny_condition_reached();
-                if should_allow {
-                    state.remove_from_denylist().await;
+                match (state.denied, state.deny_condition_reached()) {
+                    // Maybe allow the sender right after the potential RAV request. This way, the
+                    // sender can be allowed again as soon as possible if the RAV was successful.
+                    (true, false) => state.remove_from_denylist().await,
+                    // if couldn't remove from denylist, resend the message in 30 seconds
+                    // this will trigger another rav request
```

Comment:
Won't necessarily trigger another RAV request. Depends how much unaggregated fees we have. 
For ex. if the sender is denied because of low escrow and not unaggregated fees, the next call to `UpdateReceiptFees` will not a trigger a RAV.
```suggestion
                    // if couldn't remove from denylist, resend the message in 30 seconds
                    // this may trigger another rav request
```